executor/src/witgen/machines/fixed_lookup_machine.rs

```diff
@@ -207,11 +208,12 @@ impl<T: FieldElement> FixedLookup<T> {
         }
 
         // get the values of the fixed columns
-        let right = right
+        let mut right = right
             .expressions
             .iter()
-            .map(try_to_simple_poly_ref)
-            .collect::<Option<Vec<_>>>()?;
+            .filter_map(try_to_simple_poly_ref)
+            .peekable();
+        right.peek()?;
```

Comment:
Can you explain what the purpose of this peek call is?

Reply:
The previous behaviour was:
- collect an iterator of Option<T> into an Option<Vec<T>>, then early return if None
Here, we use peekable to test that the iterator is non-empty without consuming.
Peek is also needed in the internal function, which relied on Vec::first() for a few tests.

Reply:
Can you add a short comment in front of that line?